src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/Handlers/TransportCreation/TransportCreationHandler.cs

```diff
@@ -0,0 +1,215 @@
+        {
+            if (context.DataSource is null)
+            {
+                context.Error = new ArgumentNullException(nameof(context));
+                return;
+            }
+
+            try
+            {
+                // @TODO: Build CoR for handling the different protocols in order
+                if (context.DataSource.ResolvedProtocol is DataSource.Protocol.TCP)
+                {
+                    context.ConnectionStream = await HandleTcpRequest(context, isAsync, ct).ConfigureAwait(false);
+                }
+            }
+            catch (Exception e)
+            {
+                context.Error = e;
+                return;
+            }
+
+            if (NextHandler is not null)
+            {
+                await NextHandler.Handle(context, isAsync, ct).ConfigureAwait(false);
+            }
+        }
+
+        private ValueTask<Stream> HandleNamedPipeRequest()
+        {
+            throw new NotImplementedException();
+        }
+
+        private ValueTask<Stream> HandleSharedMemoryRequest()
+        {
+            throw new NotImplementedException();
+        }
+
+        private async ValueTask<Stream> HandleTcpRequest(ConnectionHandlerContext context, bool isAsync, CancellationToken ct)
+        {
+            ct.ThrowIfCancellationRequested();
+
+            // DNS lookup
+            var ipAddresses = isAsync
+                ? await Dns.GetHostAddressesAsync(context.DataSource.ServerName, ct).ConfigureAwait(false)
+                : Dns.GetHostAddresses(context.DataSource.ServerName);
+            if (ipAddresses.Length == 0)
+            {
+                throw new SocketException((int)SocketError.HostNotFound);
+            }
+
+            // If there is an IP version preference, apply it
+            switch (context.IpAddressPreference)
+            {
+                case SqlConnectionIPAddressPreference.IPv4First:
+                    Array.Sort(ipAddresses, IpAddressVersionSorter.InstanceV4);
+                    break;
+
+                case SqlConnectionIPAddressPreference.IPv6First:
+                    Array.Sort(ipAddresses, IpAddressVersionSorter.InstanceV6);
+                    break;
+
+                case SqlConnectionIPAddressPreference.UsePlatformDefault:
+                default:
+                    // Not sorting necessary
+                    break;
+            }
+
+            // Attempt to connect to one of the matching IP addresses
+            // @TODO: Handle opening in parallel
+            Socket socket = null;
+            var socketOpenExceptions = new List<Exception>();
+
+            var portToUse = context.DataSource.ResolvedPort < 0
+                ? context.DataSource.Port
+                : context.DataSource.ResolvedPort;
+            var ipEndpoint = new IPEndPoint(IPAddress.None, portToUse); // Allocate once
+            foreach (var ipAddress in ipAddresses)
```

Comment:
I get a suggestion in IDE to change to explicit type instead of var. 
Not sure if its a IDE specific suggestion or coming from .editorconfig. 

This is the rule being followed https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0007-ide0008

Can we use var when we are declaring an object, but when the object is derived from a function/iteration over enum etc, can we use explicit types (essentially what the IDE rule at the link above says)

Reply:
As far as I can tell, that doc doesn't actually assert any preference for one versus the other, just lists what the style rules are and what the true/false values mean.

I've gone back and forth on which to use over the years. Originally I was very against var, then I got retrained to use var and started to prefer it, and then C# 9 or whatever came out and now it's basically just a thing to argue over. I can adopt the suggestion, tho.